apps/mull-api/src/app/channel/channel.service.spec.ts

```diff
@@ -1,66 +1,148 @@
   let service: ChannelService;
-  let repository: MockType<Repository<Channel>>;
+  let directMessageRepository: MockType<Repository<DirectMessageChannel>>;
+  let eventRepository: MockType<Repository<EventChannel>>;
 
   beforeEach(async () => {
     const module: TestingModule = await Test.createTestingModule({
       providers: [
         ChannelService,
         { provide: getRepositoryToken(Channel), useFactory: mockChannelRepository },
+        { provide: getRepositoryToken(EventChannel), useFactory: mockEventChannelRepository },
+        {
+          provide: getRepositoryToken(DirectMessageChannel),
+          useFactory: mockDirectMessageChannelRepository,
+        },
       ],
     }).compile();
 
     service = module.get<ChannelService>(ChannelService);
-    repository = module.get(getRepositoryToken(Channel));
+    directMessageRepository = module.get(getRepositoryToken(DirectMessageChannel));
+    eventRepository = module.get(getRepositoryToken(EventChannel));
   });
 
   it('should be defined', () => {
     expect(service).toBeDefined();
   });
 
-  it('should get a channel', async () => {
-    const getChannel = await service.getChannel(mockAllChannels[0].id);
-    expect(getChannel).toBe(mockAllChannels[0]);
+  it('should get a generic channel', async () => {
+    const returnedChannel = await service.getChannel(mockAllChannels[0].id);
+    expect(returnedChannel).toBe(mockAllChannels[0]);
+  });
+
+  it('should get an event channel', async () => {
+    const returnedEventChannel = await service.getEventChannel(mockAllEventChannels[0].id);
+    expect(returnedEventChannel).toBe(mockAllEventChannels[0]);
+  });
+
+  it('should get a direct message channel', async () => {
+    const returnedDirectMessageChannel = await service.getDirectMessageChannel(
+      mockAllDirectMessageChannels[0].id,
+      mockAllUsers[0].id
+    );
+    expect(returnedDirectMessageChannel).toBe(mockAllDirectMessageChannels[0]);
+  });
+
+  it('should create a DM channel', async () => {
+    directMessageRepository.createQueryBuilder.mockReturnValue({
+      where: jest.fn().mockReturnThis(),
+      leftJoinAndSelect: jest.fn().mockReturnThis(),
+      getMany: jest.fn().mockReturnValue(mockAllDirectMessageChannels),
+    });
+    const createdDirectMessageChannel = await service.createDirectMessageChannel(
+      mockAllUsers[0].id,
+      mockAllUsers[2].id
+    );
+    expect(createdDirectMessageChannel).toBeTruthy;
+  });
+
+  it('should throw an exception when creating a DM channel with 2 users that already have a DM channel', async () => {
+    directMessageRepository.createQueryBuilder.mockReturnValue({
+      where: jest.fn().mockReturnThis(),
+      leftJoinAndSelect: jest.fn().mockReturnThis(),
+      getMany: jest.fn().mockReturnValue(mockAllDirectMessageChannels),
+    });
+    expect(
+      async () => await service.createDirectMessageChannel(mockAllUsers[0].id, mockAllUsers[1].id)
+    ).rejects.toThrow(UnauthorizedException);
```

Comment:
Some error messages resulted from this test. To fix before merging
```suggestion
    ).rejects.toThrow(ConflictException);
```